proxy/http/HttpSessionManager.cc

```diff
@@ -326,21 +326,17 @@ HttpSessionManager::acquire_session(Continuation * /* cont ATS_UNUSED */, sockad
           UnixNetVConnection *server_vc = dynamic_cast<UnixNetVConnection *>(to_return->get_netvc());
           if (server_vc) {
             UnixNetVConnection *new_vc = server_vc->migrateToCurrentThread(sm, ethread);
-            // The VC moved, free up the original one
-            if (new_vc != server_vc) {
-              ink_assert(new_vc == nullptr || new_vc->nh != nullptr);
-              if (!new_vc) {
-                // Close out to_return, we were't able to get a connection
-                to_return->do_io_close();
-                to_return = nullptr;
-                retval    = HSM_NOT_FOUND;
-              } else {
-                // Keep things from timing out on us
-                new_vc->set_inactivity_timeout(new_vc->get_inactivity_timeout());
-                to_return->set_netvc(new_vc);
-              }
+            if (new_vc->thread != ethread) {
+              // Failed to migrate, put it back to global session pool
+              m_g_pool->releaseSession(to_return);
```

Comment:
I don't seeing doing the release buying us much, but it doesn't cost us much either.  If the releaseSession doesn't get the lock, it will close the netvc anyway.

Reply:
The m_g_pool is locked at L312.
```
311     EThread *ethread       = this_ethread();
312     ProxyMutex *pool_mutex = (TS_SERVER_SESSION_SHARING_POOL_THREAD == sm->t_state.http_config_param->server_session_sharing_poo    l) ?
313                                ethread->server_session_pool->mutex.get() :
314                                m_g_pool->mutex.get();
315     MUTEX_TRY_LOCK(lock, pool_mutex, ethread);
316     if (lock.is_locked()) {
```

The ```to_return``` will be released to the pool.